lib/api/BackbeatAPI.js

```diff
@@ -101,31 +101,31 @@ class BackbeatAPI {
             ingestionSites: this._ingestionSites,
             internalStart: this._internalStart,
         };
-        const metrics = new Metrics(metricsConfig, logger);
+        this._metrics = new Metrics(metricsConfig, logger);
```

Comment:
Why is this needed?

Reply:
the objects needs to be 'stored' so the resources can be closed (in order not to leak) at the end of the tests:

```
bbapi._metrics.disconnect();
```